### PR TITLE
Remove legacy resource and redirection conversion

### PR DESCRIPTION
**Pull request by yufeih**
This PR moves legacy resource and redirection logic, to `BuildResource` and `BuildRedirection`. Legacy page and TOC conversion will be moved later.

### CODE REVIEW

**Comment by OsmondJiang** [comment]
#3950

File: src/docfx/build/legacy/files/LegacyPage.cs [hunk 1]
```diff
@@ -14,17 +14,9 @@ internal static class LegacyPage
             Document doc,
             LegacyManifestItem legacyManifestItem)
         {
-            // OPS build use TOC ouput as data page
```
Comment (live1206) [inline src/docfx/build/legacy/files/LegacyPage.cs]: keep legacy TOC in this PR? legacy logic not added to BuildTableOfContents

File: src/docfx/build/legacy/files/LegacyPage.cs [hunk 2]
```diff
@@ -44,17 +36,9 @@ internal static class LegacyPage
 
                 var themesRelativePathToOutputRoot = "_themes/";
 
-                if (!string.IsNullOrEmpty(doc.RedirectionUrl))
-                {
-                    rawMetadata = LegacyMetadata.GenerateLegacyRedirectionRawMetadata(docset, pageModel);
```
Comment (live1206) [inline src/docfx/build/legacy/files/LegacyPage.cs]: remove `LegacyMetadata.GenerateLegacyRedirectionRawMetadata()`, not in use any more

Comment (yufeih) [reply src/docfx/build/legacy/files/LegacyPage.cs]: These lines are covered in LegacyTOC, I think they are kept here accidentally during refactor.

File: src/docfx/build/legacy/files/LegacyPage.cs (hunk 1 above)
Comment (yufeih) [reply src/docfx/build/legacy/files/LegacyPage.cs]: These lines are covered in LegacyTOC, I think they are kept here accidentally during refactor.

Comment (OsmondJiang) [reply src/docfx/build/legacy/files/LegacyPage.cs]: impact test will tell us

**Comment by OsmondJiang** [comment]
redirection and resource are not using JINT to run template, can we add spec test here?

**Comment by yufeih** [comment]
> redirection and resource are not using JINT to run template, can we add spec test here?

The legacy test I added has covered both image and redirections

**Pull request merged**